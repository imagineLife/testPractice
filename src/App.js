import React from "react";
import ReactDOM from "react-dom";
const App = () => {
  return (
    <div>
      <p>Dummy React Component here!</p>
    </div>
  );
};
export default App;
ReactDOM.render(<App />, document.getElementById("app"));