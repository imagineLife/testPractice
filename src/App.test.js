import React from "react"
import ReactDOM from "react-dom"
import App from "./App"
import { configure, shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

configure({ adapter: new Adapter() })

describe("<App />", () => {
  it("should render App", () => {
    const wrapper = shallow(<App />)
  })

  //finds provided selector, <p>
  it("should have 1 paragraph", () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('p').length).toBe(1)
  })

  //checks for existing by class
  it("an app-into classed element should exist", () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('.App-intro').exists()).toBe(true)
  })

  //finds provided selector, <p>
  it("has a ul with 3 li's", () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('ul').children().length).toBe(3)
  })

  //checks that an item has a class
  it("has a ul with a class of myClass", () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('ul').hasClass('myClass')).toBe(true)
  })


  /*
	SELECTOR TYPE REVIEW
  */

  //by element type
  describe("By Selector Types", () => {
  	  

  	  it("ELEMENT with correct Text", () => {
	    const wrapper = shallow(<App />)
	    expect(wrapper.find('h1').text()).toBe('Testing Overview')
	  })

  	  it("CLASS with correct Text", () => {
	    const wrapper = shallow(<App />)
	    expect(wrapper.find('.App-title').text()).toBe('Testing Overview')
	  })	  

  	  it("ID with correct Text", () => {
	    const wrapper = shallow(<App />)
	    expect(wrapper.find('#h1ID').text()).toBe('Testing Overview')
	  })
  })
})