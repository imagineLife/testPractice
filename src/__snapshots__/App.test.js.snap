// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> works with SNAPSHOTS matches snapshot 1`] = `
<Fragment>
  <header
    className="App-header"
  >
    <h1
      alt="dummyAlt"
      className="App-title"
      id="h1ID"
    >
      Testing Overview
    </h1>
  </header>
  <Title
    text="dummy title"
  />
  <ul
    className="myClass"
  >
    <li>
      List Item 1
    </li>
    <li>
      List Item 2
    </li>
    <li>
      List Item 3
    </li>
  </ul>
  <p
    className="App-intro"
  >
    Mic Check?!
  </p>
</Fragment>
`;
